pymodbus/server/async_io.py

```diff
@@ -472,7 +470,7 @@ def __init__(
             True,
         )
 
-        self.loop = asyncio.get_running_loop()
```

Comment:
This looks wrong, it creates a new loop, the loop must be created outside the class.

Reply:
It looks like that an event loop is created (outside) in pymodbus/repl/server/main.py and before my patch is was set as an explicit parameter when creating the servers:

```
    app = ReactiveServer.factory(
        modbus_server,
        framer,
        modbus_port=modbus_port,
        slave=modbus_slave_id,
        loop=loop,
        single=False,
        data_block_settings=data_block_settings,
        **web_app_config,
        **modbus_config,
    )
```
- ModbusSerialServer was indeed using it (via evaluating kwargs.get("loop") )
- ModbusTcpServer complaint about an superfluous parameter (didn't use the event loop)
- ModbusUdpServer complaint about an superfluous "loop" parameter and I removed it, it tried to use "get_running_loop()" but it didn't work either because there was no running loop

Should the following work: as before, use the loop parameter for the Modbus{Udp,Tcp,Tls,Serial}Server, make it explicit and in case the specific server doesn't need it (looks like that the TCP-based classes don't use it), just ignore the parameter?

Reply:
I just pushed 2 commits which would implement the variant described above. Please have a look - any feedback is welcome.

Reply:
> This looks wrong, it creates a new loop, the loop must be created outside the class.

Fixed in latest commits.